.travis.yml

```diff
@@ -11,10 +11,17 @@ rust:
   - stable
   - nightly
 
+env:
+  - VERSION="stable"
+  - VERSION="nightly"
+
 matrix:
   fast_finish: true
-  allow_failures:
+  exclude:
     - rust: nightly
+      env: VERSION="stable"
+    - rust: stable
+      env: VERSION="nightly"
 
 script:
-  - cargo test --no-default-features
+  - ./build.sh
```

Comment:
Let's try and keep all the shell commands in the `script` section rather than adding `build.sh`. The build.sh script can be

```yml
- if [[ $VERSION == "nightly" ]]; then cargo test --no-default-features --features "clippy"; fi
- if [[ $VERSION != "nightly" ]]; then cargo test --no-default-features; fi
```